### PR TITLE
927 merge notif improvements

### PR DESCRIPTION
**Pull request by drewblount**
Minor improvement to a merge request notification: Adds the merge deadline to the notification, ISO-formatted.

This branch is a little complicated bc howard and I duplicated some work on individual names in the merge notification. I've rebased to keep his work and remove mine.

### CODE REVIEW

File: app/modules/individuals/models.py
Comment (drewblount) [inline app/modules/individuals/models.py:314]: This was mostly to get some tests passing without reworking them, was hitting some null issues on existing tests with individuals with no primary names.

File: app/modules/notifications/models.py
Comment (drewblount) [inline app/modules/notifications/models.py:225]: These target fields are arguably redundant and according to slack, should not be displayed to the user. Chose to keep them in to respect the frontend freeze/avoid possibly mucking up existing FE work.

File: app/modules/individuals/models.py
Comment (drewblount) [reply app/modules/individuals/models.py:314]: ("this" meaning the final else case where we return the guid)

Comment (Atticus29) [reply app/modules/individuals/models.py:314]: Hm. I could see this being an issue on FE. There are some places where we populate with "Unnamed individual" if getting a name fails.
But, if this is the only way to get tests to pass, I suppose that we could eventually check against the guid in the "Unnamed individual" logic.

Comment (Atticus29) [reply app/modules/individuals/models.py:314]: ... is it possible in codex to have an individual without a primary name??
Maybe the tests are the problem?

File: app/modules/notifications/models.py
Comment (Atticus29) [inline app/modules/notifications/models.py:233]: I'm seeing a lot of foo.isoformat() + '+00:00' (particularly on tests related to individuals) and foo.isoformat() + 'Z' in the code base and vaguely remember this being important. Might be worth checking with someone who acutally knows whether this change is needed in this case.

**Comment by Atticus29** [comment]
In case you need FE approval, here it is!

File: app/modules/individuals/models.py
Comment (drewblount) [reply app/modules/individuals/models.py:314]: This particular get_primary_name method is only called in the notification builder so I'm guessing FE is categorizing those unnamed individuals by other means.

Comment (hwindsor) [reply app/modules/individuals/models.py:314]: I'd agree that this is a more robust handling of this scenario as it is returning 'something' I'm just wondering if an empty string would be a better somethinng.

File: app/modules/individuals/models.py
Comment (hwindsor) [inline app/modules/individuals/models.py:624]: I think the outcome from the discussoin with Mark was that this should not be done. Instead what is in my PR should be. 
If this is undone then I think the changes should avoid clashing

**Comment by drewblount** [comment]
Closed this to resolve issues with Howard's PR and prevent merging into main beforehand

**Comment by drewblount** [comment]
I will manually rebase this over 961 after that's merged and reopen this. The changes to individual list on this branch will be ignored but the deadline will be kept.

**Comment by bluemellophone** [comment]
@drewblount #633 has been merged

File: app/modules/notifications/models.py
Comment (hwindsor) [reply app/modules/notifications/models.py:249]: This is a valid question and I think it's correct that something should be added but there is a mix in the code where sometimes it is ad sometimes it isn't. @naknomum can usually be relied upon to have an opinion about this :)

**Review by hwindsor — COMMENTED** [review]
LGTM but not approving yet until @naknomum responds to TZ discussion

File: app/modules/notifications/models.py
Comment (drewblount) [reply app/modules/notifications/models.py]: After convo w Jon, added 'Z' to the string. Thanks fellas!

File: app/modules/individuals/models.py
Comment (naknomum) [reply app/modules/individuals/models.py:314]: kinda favoring `None` just based on old-world trauma from there being _both_ empty-string values _and_ nulls all over the place, meaning kind of the same thing.

**Review by naknomum — APPROVED** [review]
lgtm

**Pull request merged**
